pkg/config/validation/validation.go

```diff
@@ -2360,7 +2418,12 @@ func appendErrors(err error, errs ...error) error {
 	}
 
 	for _, err2 := range errs {
-		err = appendError(err, err2)
+		switch t := err.(type) {
+		case Validation:
+			err = appendError(err, t.Err)
```

Comment:
will this case also return `t.Warning`?

Reply:
No, if you use appendError it will drop the warning. The alternative is it will be broken if someone calls appendError on a Validation (its really easy to do this and hard to debug) or appendValidation cannot accept both types. We cannot make appendValidation accept `error|Validation` without making `Validation` an `error` as well, which impacts appendError